src/compiler/cpp_generator.cc

```diff
@@ -993,7 +995,23 @@ void PrintHeaderServerMethodCallback(
         "controller) {\n"
         "               return this->$"
         "Method$(context, request, response, controller);\n"
-        "             }));\n");
+        "             }, nullptr));\n}\n");
+    printer->Print(
+        *vars,
+        "void SetMessageAllocatorFor_$Method$(\n"
+        "    ::grpc::MessageAllocator<$RealRequest$, $RealResponse$>* "
+        "allocator) {\n"
+        "  ::grpc::Service::experimental().MarkMethodCallback($Idx$,\n"
+        "    new ::grpc::internal::CallbackUnaryHandler< "
+        "$RealRequest$, $RealResponse$>(\n"
+        "      [this](::grpc::ServerContext* context,\n"
+        "             const $RealRequest$* request,\n"
+        "             $RealResponse$* response,\n"
+        "             ::grpc::experimental::ServerCallbackRpcController* "
+        "controller) {\n"
+        "               return this->$"
+        "Method$(context, request, response, controller);\n"
+        "             }, allocator));\n");
```

Comment:
This will soon also have another variant based on #18773 ...

Reply:
Actually the new variant probably makes this unusable, so perhaps this should be added as a setter to CallbackUnaryHandler rather than overriding the allocated CallbackUnaryHandler altogether?